server/pulp/server/webservices/views/content.py

```diff
@@ -1,13 +1,378 @@
+        Deletes all orphans of the specified type.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param content_type: restrict the list of orphans to be deleted to this content type
+        :type  content_type: string
+
+        :raises: OperationPostponed
+        """
+        orphan_manager = factory.content_orphan_manager()
+        task_tags = [tags.resource_tag(tags.RESOURCE_CONTENT_UNIT_TYPE, 'orphans')]
+        async_task = orphan_manager.delete_orphans_by_type.apply_async(
+            (content_type,), tags=task_tags
+        )
+
+        raise OperationPostponed(async_task)
+
+
+class OrphanResourceView(View):
+    """
+    Views for a specific orphan.
+    """
+    @auth_required(authorization.READ)
+    def get(self, request, content_type, unit_id):
+        """
+        Return a serialized object representing the requested orphan
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param content_type: content type of the requested orphan
+        :type  content_type: string
+        :param unit_id: id of the requested unit
+        :type  unit_id: string
+
+        :return: response conainting a serialized dict of the requested orphan
+        :rtype : django.http.HttpResponse
+        """
+        orphan_manager = factory.content_orphan_manager()
+        orphan_dict = orphan_manager.get_orphan(content_type, unit_id)
+        orphan_dict['_href'] = request.get_full_path()
+
+        return generate_json_response(orphan_dict)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, content_type, unit_id):
+        """
+        Delete a specified orphan.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param content_type: content type of the requested orphan
+        :type  content_type: string
+        :param unit_id: id of the requested orphan
+        :type  unit_id: string
+
+        :raises: OperationPostponed
+        """
+        orphan_manager = factory.content_orphan_manager()
+        unit_info = [{'content_type_id': content_type, 'unit_id': unit_id}]
+        task_tags = [tags.resource_tag(tags.RESOURCE_CONTENT_UNIT_TYPE, 'orphans')]
+        async_task = orphan_manager.delete_orphans_by_id.apply_async((unit_info,), tags=task_tags)
+
+        raise OperationPostponed(async_task)
+
+
+class DeleteOrphansActionView(View):
+    """
+    Deprecated in 2.4, please use the more restful OrphanResource delete instead
+
+    Delete all orphans.
+    """
+
+    @auth_required(authorization.DELETE)
+    def post(self, request):
+        """
+        Pass the appropriate tags and json object to delete_orphans_by_id.
+        Raises an OperationPostponed.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :raises: OperationPostponed
```

Comment:
Maybe indicate that this is the normal codepath? It's a little strange that the normal exit codepath is an exception. Probably make a similar change throughout the PR.